src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,108 @@
+package duke.task;
+
+import duke.exception.TaskAlreadyCompletedException;
+import duke.exception.TaskNotExistException;
+
+import java.util.ArrayList;
+
+public class Task {
+    private boolean isDone;
+    private String description;
+    private static int taskCounter = 0;
+    private static int completedTaskCounter = 0;
+    private static ArrayList<Task> tasks = new ArrayList<>();
+    private static Task recentTask;
+
+    public Task(String description){
+        this.description = description;
+        // default status for isDone is false
+        isDone = false;
+    }
+
+    public boolean isDone() {
+        return isDone;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public void markAsDone(){
+        this.isDone = true;
+    }
+
+    public String getStatusIcon(){
+        return (isDone ? "[ \u2713 ] " : "[ \u2718 ] ");
+    }
+
+    // static methods for all the Tasks
+    public static int getTaskCounter() {
+        return taskCounter;
+    }
+
+    public static int getRemainingTasks() {
+        return taskCounter - completedTaskCounter;
+    }
+
+    public static ArrayList<Task> getTaskList() {
+        return tasks;
+    }
+
+    public static void addNewTask(Task newTask) {
+        tasks.add(newTask);
+        incrementTaskCounter();
+    }
+
+    public static void deleteTask(int taskNumber){
+        recentTask = tasks.get(taskNumber);
+        if (tasks.get(taskNumber).isDone) {
+            completedTaskCounter--;
+        }
+        tasks.remove(taskNumber);
+        System.out.println("The deleted task is : " + recentTask);
+        taskCounter--;
+    }
+
+
+    public static void completeTask(int taskNumber) {
+        tasks.get(taskNumber).markAsDone();
+        recentTask = tasks.get(taskNumber);
+        // completedTaskIndex = taskNumber;
```

Comment:
Could consider deleting dead code. Refer to code quality.

Reply:
Done! Thank you!